human-forecasts/server-distribution-code.R

```diff
@@ -0,0 +1,432 @@
+
+update_values <- function(horizon = NULL) {
+  
+  if (is.null(horizon)) {
+    steps <- 1:4
+  } else {
+    steps <- horizon
+  }
+  
+  rv$median <- rv$median_latent
+  rv$width <- rv$width_latent
+  
+  for (i in steps) {
+    
+    # rv[[paste0("forecasts_week_", i)]] <<- qlnorm(quantile_grid, 
+    #                                               meanlog = log(rv$median[i]), 
+    #                                               sdlog = as.numeric(rv$width[i]))
+    
+    if (input$distribution == "log-normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- exp(qnorm(quantile_grid, 
+                                                       mean = log(rv$median[i]),
+                                                       sd = as.numeric(rv$width[i])))
+    } else if (input$distribution == "normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- (qnorm(quantile_grid, 
+                                                    mean = (rv$median[i]),
+                                                    sd = as.numeric(rv$width[i])))
+      
+    } else if (input$distribution == "cubic-normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- (qnorm(quantile_grid, 
+                                                    mean = (rv$median[i]) ^ (1 / 3),
+                                                    sd = as.numeric(rv$width[i]))) ^ 3
+    } else if (input$distribution == "fifth-power-normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- (qnorm(quantile_grid, 
+                                                    mean = (rv$median[i]) ^ (1 / 5),
+                                                    sd = as.numeric(rv$width[i]))) ^ 5
+    } else if (input$distribution == "seventh-power-normal") {
```

Comment:
Suggest using. Helper fun to avoid copy pasting.

Reply:
how could that function look like? I could wrap all this in a helper function, but can I get rid of the if else structure?